### PR TITLE
[MI-3215] Added testcases for getMentionsData

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**

### CODE REVIEW

File: server/message_hooks_test.go [hunk 1]
```diff
@@ -1995,3 +1995,185 @@ func TestGetChatIDForChannel(t *testing.T) {
...
+		Message               string
+		ChatID                string
+		SetupAPI              func(*plugintest.API)
+		SetupStore            func(*storemocks.Store)
+		SetupClient           func(*clientmocks.Client)
+		ExpectedMessage       string
+		ExpectedMentionsCount int
+	}{
+		{
+			Name:            "GetMentionsData: mentioned all outside a channel",
```
Comment (ayusht2810) [inline server/message_hooks_test.go:2011]:
```suggestion
			Name:            "GetMentionsData: mentioned in direct chat message",
```

File: server/message_hooks_test.go [hunk 2]
```diff
@@ -1995,3 +1995,185 @@ func TestGetChatIDForChannel(t *testing.T) {
...
+			ExpectedMessage: "Hi <at id=\"0\"></at>",
+			SetupAPI:        func(api *plugintest.API) {},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChannelInTeam", testutils.GetTeamID(), testutils.GetChannelID()).Return(&msteams.Channel{}, nil)
+			},
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: error occurred while getting channel in team",
```
Comment (ayusht2810) [inline server/message_hooks_test.go:2062]:
```suggestion
			Name:            "GetMentionsData: error occurred while getting MS Teams channel",
```

File: server/message_hooks_test.go [hunk 3]
```diff
@@ -1995,3 +1995,185 @@ func TestGetChatIDForChannel(t *testing.T) {
...
+			Message:         "Hi @all",
+			ExpectedMessage: "Hi <at id=\"0\">@all</at>",
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", "Unable to get ms teams channel", "Error", mock.Anything)
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChannelInTeam", testutils.GetTeamID(), testutils.GetChannelID()).Return(nil, errors.New("error occurred while getting channel in team"))
```
Comment (ayusht2810) [inline server/message_hooks_test.go:2069]: same

File: server/message_hooks_test.go [hunk 4]
```diff
@@ -1995,3 +1995,185 @@ func TestGetChatIDForChannel(t *testing.T) {
...
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: mentioned a user",
+			Message:         "Hi @test-username",
+			ExpectedMessage: "Hi <at id=\"0\"></at>",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetUserByUsername", "test-username").Return(&model.User{
```
Comment (ayusht2810) [inline server/message_hooks_test.go:2080]: there is a testutil present for this, check if we can use that here. fix everywhere, if updating

File: server/message_hooks_test.go [hunk 5]
```diff
@@ -1995,3 +1995,185 @@ func TestGetChatIDForChannel(t *testing.T) {
...
+					Id: testutils.GetID(),
+				}, nil)
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetUser", testutils.GetUserID()).Return(&msteams.User{}, nil)
```
Comment (ayusht2810) [inline server/message_hooks_test.go:2085]: i think we should fill in this data, as the expected message will change on these values. fix everyhwere

File: server/message_hooks_test.go [hunk 6]
```diff
@@ -1995,3 +1995,185 @@ func TestGetChatIDForChannel(t *testing.T) {
...
+			SetupStore:            func(store *storemocks.Store) {},
+			ExpectedMentionsCount: 1,
+		},
+		{
+			Name:            "GetMentionsData: mentioned all but chat ID is missing",
+			Message:         "Hi @all",
+			ExpectedMessage: "Hi <at id=\"0\"></at>",
+			SetupAPI:        func(api *plugintest.API) {},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChannelInTeam", testutils.GetTeamID(), testutils.GetChannelID()).Return(&msteams.Channel{}, nil)
```
Comment (ayusht2810) [inline server/message_hooks_test.go:2056]: fill this data as well

File: server/message_hooks_test.go [hunk 7]
```diff
@@ -1995,3 +1995,185 @@ func TestGetChatIDForChannel(t *testing.T) {
...
+			ExpectedMentionsCount: 2,
+		},
+		{
+			Name:            "GetMentionsData: error getting MM user with username",
+			Message:         "Hi @test-username",
+			ExpectedMessage: "Hi @test-username",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", "Unable to get user by username", "Error", mock.Anything)
+				api.On("GetUserByUsername", "test-username").Return(nil, &model.AppError{
```
Comment (ayusht2810) [inline server/message_hooks_test.go:2120]: use already present testutil here

File: server/message_hooks_test.go [hunk 8]
```diff
@@ -1995,3 +1995,177 @@ func TestGetChatIDForChannel(t *testing.T) {
...
+			Name:            "GetMentionsData: mentioned all and a specific user in a channel",
+			Message:         "Hi @all @test-username",
+			ExpectedMessage: "Hi <at id=\"0\">Everyone</at> <at id=\"1\">mock-name</at>",
+			ChatID:          testutils.GetChatID(),
+			SetupAPI: func(api *plugintest.API) {
+				api.On("GetUserByUsername", "test-username").Return(testutils.GetUser("mock-role", "mock-email"), nil)
+			},
+			SetupClient: func(client *clientmocks.Client) {
+				client.On("GetChat", testutils.GetChatID()).Return(&msteams.Chat{
+					Type: "G",
```
Comment (Kshitij-Katiyar) [inline server/message_hooks_test.go:2103]: check if there is a constant present for this

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by ayusht2810 — APPROVED** [review]

File: server/message_hooks_test.go (hunk 8 above)
Comment (raghavaggarwal2308) [reply server/message_hooks_test.go:2103]: Not present

**Pull request merged**